chains/evm/voter/message-handler.go

```diff
@@ -70,40 +71,52 @@ func (mh *EVMMessageHandler) RegisterMessageHandler(address string, handler Mess
 }
 
 func ERC20MessageHandler(m *message.Message, handlerAddr, bridgeAddress common.Address) (*proposal.Proposal, error) {
-	if len(m.Payload) != 2 {
-		return nil, errors.New("malformed payload. Len  of payload should be 2")
+	payloadLen := len(m.Payload)
+	if payloadLen != 2 && payloadLen != 3 {
+		return nil, errors.New("malformed payload. Len  of payload should be 2 or 3")
 	}
 	amount, ok := m.Payload[0].([]byte)
 	if !ok {
-		return nil, errors.New("wrong payloads amount format")
+		return nil, errors.New("wrong payload amount format")
 	}
 	recipient, ok := m.Payload[1].([]byte)
 	if !ok {
-		return nil, errors.New("wrong payloads recipient format")
+		return nil, errors.New("wrong payload recipient format")
 	}
 	var data []byte
 	data = append(data, common.LeftPadBytes(amount, 32)...) // amount (uint256)
 	recipientLen := big.NewInt(int64(len(recipient))).Bytes()
 	data = append(data, common.LeftPadBytes(recipientLen, 32)...) // length of recipient (uint256)
 	data = append(data, recipient...)                             // recipient ([]byte)
+
+	if payloadLen == 3 {
+		priority, ok := m.Payload[2].([]byte)
+		if !ok {
+			return nil, errors.New("wrong payload priority format")
+		}
+		priorityLen := big.NewInt(int64(len(priority))).Bytes()
+		data = append(data, common.LeftPadBytes(priorityLen, 1)...) // length of priority (uint8)
```

Comment:
Does data have to contain priority data? Seems irrelevant for submitting on chain.. 
I would probably add https://www.sohamkamani.com/golang/options-pattern/ (`...options`) param to creating new proposal and add all extra options that just the relayer needs.